packages/angular/src/components/search/search-button.ts

```diff
@@ -11,44 +17,57 @@ import { Button } from '../button'
  * @param {string} [aria-label] - Aria label for the button
  *
  */
-@Component({
-  selector: 'button[ksd-search-button]',
+@Directive({
+  selector: 'button[ksdSearchButton]',
   standalone: true,
-  template: ` <ng-content>Søk</ng-content> `,
   host: {
     type: 'submit',
     '[attr.aria-label]': 'this.ariaLabel()',
     '[attr.variant]': 'this.variant()',
   },
 })
-export class SearchButton extends Button implements OnInit {
+export class SearchButton {
+  private readonly button = inject(ElementRef<Button>)
+
   /**
    * Specify which button variant to use
    *
-   * Note: \'tertiary\' variant is not supported, but needed to be accepted
-   * as input to avoid compiler errors for narrowing types.
+   * Note: Since the `SearchButton` is an instance of `Button`,
+   * but should only allow `'primary'` and `'secondary'` variants.
+   * `'tertiary'` will compile but will log a warning in dev mode.
+   *
    * @default 'primary'
    *
    */
-  override readonly variant = input<'primary' | 'secondary' | 'tertiary'>(
-    'primary',
-  )
+  readonly variant = input<'primary' | 'secondary'>('primary')
 
   /**
    * Aria label for the button
    */
-  protected readonly ariaLabel = input('', { alias: 'aria-label' })
+  readonly ariaLabel = input('', { alias: 'aria-label' })
 
   /**
-   * Runtime check for unsupported 'tertiary' variant
+   * Check that component is a ksd-button at runtime
    */
-  ngOnInit(): void {
-    if (this.variant() === 'tertiary') {
-      logIfDevMode({
-        component: 'SearchButton',
-        message:
-          "The 'tertiary' variant is not supported - use 'primary' or 'secondary'.",
-      })
-    }
+  constructor() {
+    afterNextRender(() => {
```

Comment:
I think it would be cleaner to just remove this logic and just put the `ds-button` class on the host-element, so we keep them untangled